pkgs/optee/default.nix

```
@@ -18,18 +19,19 @@ let
 
   opteeClient = stdenv.mkDerivation {
     pname = "optee_client";
-    version = l4tVersion;
+    version = l4tMajorMinorPatchVersion;
     src = nvopteeSrc;
-    patches = [
-      ./0001-Don-t-prepend-foo-bar-baz-to-TEEC_LOAD_PATH.patch
-      (fetchpatch {
-        name = "tee-supplicant-Allow-for-TA-load-path-to-be-specified-at-runtime.patch";
-        url = "https://github.com/OP-TEE/optee_client/commit/f3845d8bee3645eedfcc494be4db034c3c69e9ab.patch";
-        stripLen = 1;
-        extraPrefix = "optee/optee_client/";
-        hash = "sha256-XjFpMbyXy74sqnc8l+EgTaPXqwwHcvni1Z68ShokTGc=";
-      })
-    ];
+    patches =
+      if l4tAtLeast "36" then [ ] else [
+        ./0001-Don-t-prepend-foo-bar-baz-to-TEEC_LOAD_PATH.patch
+        (fetchpatch {
+          name = "tee-supplicant-Allow-for-TA-load-path-to-be-specified-at-runtime.patch";
+          url = "https://github.com/OP-TEE/optee_client/commit/f3845d8bee3645eedfcc494be4db034c3c69e9ab.patch";
+          stripLen = 1;
+          extraPrefix = "optee/optee_client/";
+          hash = "sha256-XjFpMbyXy74sqnc8l+EgTaPXqwwHcvni1Z68ShokTGc=";
+        })
+      ];
```

Will need to add `l4tAtLeast` to `callPackage`-provided arguments.

```suggestion
    patches = lib.optionals (l4tOlder "36") [
      ./0001-Don-t-prepend-foo-bar-baz-to-TEEC_LOAD_PATH.patch
      (fetchpatch {
        name = "tee-supplicant-Allow-for-TA-load-path-to-be-specified-at-runtime.patch";
        url = "https://github.com/OP-TEE/optee_client/commit/f3845d8bee3645eedfcc494be4db034c3c69e9ab.patch";
        stripLen = 1;
        extraPrefix = "optee/optee_client/";
        hash = "sha256-XjFpMbyXy74sqnc8l+EgTaPXqwwHcvni1Z68ShokTGc=";
      })
    ];
```